addons/mail/models/mail_message.py

```diff
@@ -945,6 +945,19 @@ def _message_fetch(self, domain, *, thread=None, search_term=None, is_notificati
                 message_domain |= Domain("id", "in", accessible_tracking_value_ids.mail_message_id.ids)
             domain &= message_domain
             res["count"] = self.search_count(domain)
+
+        if self.env.context.get('is_chatter_view'):
+            if around and (around := self.browse(around)):
+                messages_before = self.search(domain & Domain('date', '<=', around.date), limit=limit // 2, order='date DESC, id DESC')
+                messages_after = self.search(domain & Domain('date', '>', around.date), limit=limit // 2, order='date ASC, id ASC')
```

Comment:
This domain would need to take into account both the "around" message `(create_)date` and `id`, as if 2 messages have the same `create_date`, we still need to check `id` to correctly take (or ignore) other messages having the same date.